### PR TITLE
Support GCP managed postgresql database

### PR DESCRIPTION
**Pull request by kfurgol**
**Description**

Changes proposed in this pull request:

- Added Google's cloudsql-proxy to director deployment, which is being installed  when `database.useEmbedded` value is set to `false`
- Added section to README.md file about using managed database

**Related issue(s)**
#80

### CODE REVIEW

File: chart/compass/charts/director/templates/deployment.yaml [hunk 1]
```diff
@@ -30,3 +30,21 @@ spec:
               value: "0.0.0.0:{{ .Values.deployment.args.containerPort }}"
             - name: APP_PLAYGROUND_API_ENDPOINT
               value: "/director/graphql"
+        {{if eq .Values.global.database.useEmbedded false}}
+        - name: cloudsql-proxy
+          image: gcr.io/cloudsql-docker/gce-proxy:1.11
+          command: ["/cloud_sql_proxy",
+                    "-instances={{ .Values.global.database.instanceConnectionName }}",
```
Comment (aszecowka) [inline chart/compass/charts/director/templates/deployment.yaml]: missing `=tcp:5432`

File: chart/compass/values.yaml [hunk 2]
```diff
@@ -41,4 +41,13 @@ global:
       namespace: compass-system
 
   database:
-    useEmbedded: true
\ No newline at end of file
+    useEmbedded: true
+    # Values for GCP managed PostgreSQL database
```
[the remark this answers is not in the document]
Comment (kfurgol) [reply chart/compass/values.yaml]: done

File: chart/compass/README.md [hunk 3]
```diff
@@ -19,3 +19,21 @@ To learn more about the Compass, see the [Overview](https://github.com/kyma-incu
...
 | --- | --- | --- | --- |
 | `database.useEmbedded` | Specifies whether `postgresql` chart should be installed | true/false | `true` |
+
+## Install the Compass with managed GCP PostgreSQL database
+
+To install the Compass with GCP managed Postgres database, set `database.useEmbedded` value to `false`, and fill those values:
+
+| Parameter | Description | Values | Default |
+| --- | --- | --- | --- |
+| `database.gcpKey` | Specifies base64 encoded key with Google credentials | base64 encoded string | "" |
```
[the remark this answers is not in the document]
Comment (kfurgol) [reply chart/compass/README.md]: done

[the remark this answers is not in the document]
Comment (kfurgol) [reply chart/compass/README.md]: done

File: chart/compass/README.md [hunk 4]
```diff
@@ -19,3 +19,21 @@ To learn more about the Compass, see the [Overview](https://github.com/kyma-incu
...
+| `database.instanceConnectionName` | Specifies instance connection name to GCP PostgreSQL database | string | "" |
+| `database.dbUser` | Specifies database username | base64 encoded string | "" |
+| `database.dbPassword` | Specifies password for database user | base64 encoded string | "" |
+| `database.dbName` | Specifies database name | string | "" |
+| `database.host` | Specifies cloudsql-proxy host (usually `localhost`) | string | "" |
+| `database.hostPort` | Specifies cloudsql-proxy port (usually `5432`) | string | "" |
+| `database.sslMode` | Specifies SSL connection mode | string | "" |
```
[the remark this answers is not in the document]
Comment (kfurgol) [reply chart/compass/README.md]: added link to possible values in the new `.md` file

File: chart/compass/charts/director/templates/deployment.yaml (hunk 1 above)
Comment (kfurgol) [reply chart/compass/charts/director/templates/deployment.yaml]: added it

File: chart/compass/README.md [hunk 5]
```diff
@@ -19,3 +19,21 @@ To learn more about the Compass, see the [Overview](https://github.com/kyma-incu
 | Parameter | Description | Values | Default |
 | --- | --- | --- | --- |
 | `database.useEmbedded` | Specifies whether `postgresql` chart should be installed | true/false | `true` |
+
+## Install the Compass with managed GCP PostgreSQL database
+
+To install the Compass with GCP managed Postgres database, set `database.useEmbedded` value to `false`, and fill those values:
```
[the remark this answers is not in the document]
Comment (kfurgol) [reply chart/compass/README.md]: done

File: chart/compass/README.md [hunk 6]
```diff
@@ -19,3 +19,21 @@ To learn more about the Compass, see the [Overview](https://github.com/kyma-incu
...
+
+| Parameter | Description | Values | Default |
+| --- | --- | --- | --- |
+| `database.gcpKey` | Specifies base64 encoded key with Google credentials | base64 encoded string | "" |
+| `database.instanceConnectionName` | Specifies instance connection name to GCP PostgreSQL database | string | "" |
+| `database.dbUser` | Specifies database username | base64 encoded string | "" |
+| `database.dbPassword` | Specifies password for database user | base64 encoded string | "" |
```
[the remark this answers is not in the document]
Comment (kfurgol) [reply chart/compass/README.md]: done

File: chart/compass/README.md [hunk 7]
```diff
@@ -19,3 +19,21 @@ To learn more about the Compass, see the [Overview](https://github.com/kyma-incu
...
+| `database.dbName` | Specifies database name | string | "" |
+| `database.host` | Specifies cloudsql-proxy host (usually `localhost`) | string | "" |
+| `database.hostPort` | Specifies cloudsql-proxy port (usually `5432`) | string | "" |
```
[the remark this answers is not in the document]
Comment (kfurgol) [reply chart/compass/README.md]: added for `host` and `hostPort`

File: chart/compass/values.yaml [hunk 8]
```diff
@@ -41,4 +41,14 @@ global:
       namespace: compass-system
 
   database:
-    useEmbedded: true
\ No newline at end of file
+    useEmbedded: false
```
Comment (aszecowka) [inline chart/compass/values.yaml]: useEmbedded should be still set to true IMO

File: docs/managed-database-connection.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
+# Managed GCP PostgreSQL database connection
```
Comment (aszecowka) [inline docs/managed-database-connection.md]: maybe rename it to Configure Managed GCP PostgreSQL

File: docs/managed-database-connection.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+| `global.database.managedGCP.dbUser` | Specifies database username | base64 encoded string | "" |
+| `global.database.managedGCP.dbPassword` | Specifies password for database user | base64 encoded string | "" |
+| `global.database.managedGCP.dbName` | Specifies database name | string | "" |
+| `global.database.managedGCP.host` | Specifies cloudsql-proxy host | string | "localhost" |
+| `global.database.managedGCP.hostPort` | Specifies cloudsql-proxy port | string | "5432" |
+| `global.database.managedGCP.sslMode` | Specifies SSL connection mode | string | "" |
+
+To connect to managed database, we use [cloudsql-proxy](https://cloud.google.com/sql/docs/postgres/sql-proxy) provided by Google, which consumes `serviceAccountKey` and `instanceConnectionName` values.
+
+Into `serviceAccountKey` value put the key for GCP Service Account mentioned in prerequisites.
```
Comment (aszecowka) [inline docs/managed-database-connection.md]: You should mention that in the table above.

File: docs/managed-database-connection.md [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+To find `Instance connection name`, go to the [SQL Instances page](https://console.cloud.google.com/sql/instances) and open desired database overview.
+![Instance connection String](./assets/sql-instances-list.png)
+
+Than look for `Instance connection name` box inside `Connect to this instance` section.
+
+![Instance connection String](./assets/instance-connection-string.png)
+
+For `dbUser` and `dbPassword` values, use one of accounts from `USERS` tab. The `dbName` value is name of database which can be found
```
Comment (aszecowka) [inline docs/managed-database-connection.md]: which can be found ...

File: chart/compass/values.yaml (hunk 8 above)
Comment (kfurgol) [reply chart/compass/values.yaml]: done

File: docs/managed-database-connection.md (hunk 9 above)
Comment (kfurgol) [reply docs/managed-database-connection.md]: done

File: docs/managed-database-connection.md (hunk 10 above)
Comment (kfurgol) [reply docs/managed-database-connection.md]: done

File: docs/managed-database-connection.md (hunk 11 above)
Comment (kfurgol) [reply docs/managed-database-connection.md]: fixed

File: docs/configure-managed-gcp-postgresql.md [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+## Install the Compass with managed GCP PostgreSQL database
+
+To install the Compass with GCP managed Postgres database, set `database.useEmbedded` value to `false` inside `./chart/compass/values.yaml` file, and fill those values:
+
+| Parameter | Description | Values | Default |
+| --- | --- | --- | --- |
+| `global.database.managedGCP.serviceAccountKey` | Specifies base64 encoded the key for GCP Service Account mentioned in prerequisites. | base64 encoded string | "" |
+| `global.database.managedGCP.instanceConnectionName` | Specifies instance connection name to GCP PostgreSQL database | string | "" |
+| `global.database.managedGCP.dbUser` | Specifies database username | base64 encoded string | "" |
```
Comment (aszecowka) [inline docs/configure-managed-gcp-postgresql.md]: it should not be base64 encoded string IMO, the same with password

Comment (kfurgol) [reply docs/configure-managed-gcp-postgresql.md]: done

**Pull request merged**